server/grpc/runner/runner.go

```diff
@@ -139,9 +143,15 @@ func (s *Server) Result(ctx context.Context, req *ResultRequest) (*ResultRespons
 	if err != nil {
 		return nil, err
 	}
+	start, ok := s.inProgress.Load(req.ExecutionHash.String())
+	if !ok {
+		panic("execution should be in memory")
```

Comment:
return error please

Reply:
The error that is returned is the error for the client (here the service). In this case, the error is an orchestrator error and should be treated by the orchestrator otherwise we might skip these kinds of errors.
I agree that the panic is a bit too much but I don't think to return the error is safe here. Maybe we should have an error channel in the orchestrator or something like that.

Reply:
let's put the error in the logger (the logger should be passed to this struct, the package `server/grpc` has it)

Reply:
done and set a default start time to the current time (that should be at least 1nano sec that will be converted to 1s of execution)